src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,163 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private MyNode head;
+    private MyNode tail;
```

Comment:
The `MyNode` class should be an inner class within `MyLinkedList` as per the checklist: 'Make `Node<T>` an inner class.'